chainerx_cc/chainerx/routines/math.cc

```diff
@@ -1005,6 +1005,42 @@ Array Arctan(const Array& x) {
     return out;
 }
 
+Array Arctan2(const Array& x1, const Array& x2) {
+    Dtype out_dtype = GetMathResultDtype(GetArithmeticResultDtype(x1, x2));
+    Array x1_cast = x1.dtype() != out_dtype ? x1.AsType(out_dtype) : x1;
+    Array x2_cast = x2.dtype() != out_dtype ? x2.AsType(out_dtype) : x2;
```

Comment:
Ditto.
```suggestion
    const Array& x2_cast = x2.dtype() != out_dtype ? x2.AsType(out_dtype) : x2;
```

Reply:
The device already casts them, so removed it from here, and backward takes care of the type of `gradient Array` similar to `AddImpl`